README.md

```diff
@@ -1,16 +1,1831 @@
+read.promise.await --> read.task: error
+read.promise.await --> read.task: *
+
+
+send.receipt.out --> send.result
+send.receipt.sig --> send.receipt.ran
+send.receipt.sig --> send.receipt.out
+send.receipt.sig --> send.receipt.fx
+send.receipt.sig --> send.receipt.meta
+send.receipt.sig --> send.receipt.iss
+send.receipt.sig --> send.receipt.prf
+
+
+send.receipt.fx --> send.fx.fork
+send.receipt.fx --> send.fx.join
+send.fx.fork --> update.task
+send.fx.fork --> read.task
+read.cause --> send
+update.cause --> send
+send.receipt.ran --> send
+```
+
+### 2.2.1 Task
+
+A [Task] is like a deferred function application: a request to perform some action on a resource with specific input.
+
+### 2.2.2 Authorization
+
+An [Authorization] is a cryptographically signed proof permitting execution of referenced tasks. It allows invoker to authorize a group of tasks using one cryptographic signature.
+
+### 2.2.3 Invocation
+
+An [Invocation] is an invoker authorized instruction to the [Executor] to run the [Task].
+
+### 2.2.4 Result
+
+A [Result] is the output of a [Task].
+
+### 2.2.5 Receipt
+
+A [Receipt] is a cryptographically signed description of the [Invocation] output.
+
+### 2.2.6 Effect
+
+An [Effect] are the instruction to the [Executor] to run set of [Task]s concurrently.
+
+## 2.3 IPLD Schema
+
+```ipldsch
+type Task struct {
+  with    URI
+  do      Ability
+  input   {String: any}
+
+  nnc     string
+}
+
+type URI string
+type Ability string
+
+type Authorization struct {
+  # Authorization is denoted by the set of links been authorized
+  scope   [&Any]
+  # Scope signed by the invoker
+  s       VarSig
+}
+
+type Invocation struct {
+  v       SemVer
+
+  run     &Task
+  # Receipt of the invocation that caused this invocation
+  cause   optional &Invocation
+  # Task authorization.
+  auth    &Authorization
+
+  meta    {String : any}
+
+  prf     [&UCAN]
+}
+type SemVer string
+
+type Receipt struct {
+  # Invocation this is a receipt for
+  ran     &Invocation
+
+  # Output of the invocation
+  out     Result
+
+  # Effects to be performed
+  fx      Effect
+
+  # All the other metadata
+  meta    {String: any}
+
+  # Principal that issued this receipt. If omitted issuer is
+  # inferred from the invocation task audience.
+  iss     optional Principal
+
+  # When issuer is different from executor this MUST hold a UCAN
+  # delegation chain from executor to the issuer. Should be omitted when
+  # executor is an issuer.
+  prf     [&UCAN]
+
+  # Signature from the "iss".
+  s       Varsig
+}
+
+type Result union {
+  | any    "ok"    # Success
+  | any    "error" # Error
+} representation kinded
+
+# Represents a request to invoke enclosed set of tasks concurrently
+type Effect struct {
+  # Primary set of tasks to be invoked
+  fork      [&Invocation]
+  
+  # Additional task to be invoked with added semantics
+  # of representing a workflow execution continuation.
+  join       optional &Promise
```

Comment:
Is it possible to avoid the word "workflow" here (because we also use "workflow" to mean a collection of tasks, which may also be represented as `fork`).

```suggestion
  # Continuation for straight-line programs
  join       optional &Promise
```